server/lomas_server/mongodb_admin.py

```diff
@@ -528,16 +528,16 @@ def add_dataset(  # pylint: disable=too-many-arguments, too-many-locals
     metadata_database_type: str,
     dataset_path: Optional[str] = "",
     metadata_path: Optional[str] = "",
-    s3_bucket: Optional[str] = "",
-    s3_key: Optional[str] = "",
+    bucket: Optional[str] = "",
+    key: Optional[str] = "",
     endpoint_url: Optional[str] = "",
-    aws_access_key_id: Optional[str] = "",
-    aws_secret_access_key: Optional[str] = "",
-    metadata_s3_bucket: Optional[str] = "",
-    metadata_s3_key: Optional[str] = "",
+    credentials_name: Optional[str] = "",
+    metadata_bucket: Optional[str] = "",
+    metadata_key: Optional[str] = "",
     metadata_endpoint_url: Optional[str] = "",
-    metadata_aws_access_key_id: Optional[str] = "",
-    metadata_aws_secret_access_key: Optional[str] = "",
+    metadata_access_key_id: Optional[str] = "",
+    metadata_secret_access_key: Optional[str] = "",
```

Comment:
I think that it mightbe better to handle metadata credentials like the file credentials (they might be in a protected storage too)

Reply:
The reason we need the credentials here is that the function in mongo_admin.py (used by mongo_admin_cli.py) need the credentials to download the metadata from s3 and add them to the admin database.

When used from the cli script, we do not have access to the server config (which is partly loaded from a kubernetes secret in the server).

The solution I propose is to create server endpoints for all functions in mongo_admin.py. Only administrator users would be able to call them. The cli script would just make calls to the server api instead of directly running the mongo_admin functions. These could then have access to the server config and the admin_database since they would be part of the server.

Reply:
Ah ok, I understand the issue now. Yes I think it is a good solution and it would also be better with the same administrator user limitation for the endpoint `"/get_memory_usage"`.

Reply:
Yes exactly. I think it also enhances the security aspect of things, and if we add another type of admin db some day, we can keep the same cli script and just change server.